### PR TITLE
Create a demo application with CMS capabilities (blogging, pages, comments etc.)

### PR DESCRIPTION
**Pull request by EngincanV**
Resolves https://github.com/abpframework/abp/issues/17555

---

## Homepage

image

## Image Gallery

image

### Image Detail with reactions & comments

image

## Products -> ABP Commercial

image

## Products -> ASPNET Zero

image

## Blog & Blogpost

image

image

## Our Team

image

### CODE REVIEW

File: src/CmsKitDemo/CmsKitDemoModule.cs [hunk 1]
```diff
@@ -321,6 +326,37 @@ private void ConfigureEfCore(ServiceConfigurationContext context)
         });
     }
 
+    private void ConfigureRazorPages()
+    {
+        Configure<RazorPagesOptions>(options =>
+        {
+            options.Conventions.AddPageRoute("/Gallery/Index", "image-gallery");
+            options.Conventions.AddPageRoute("/Gallery/Detail", "image-gallery/detail/{ImageId}");
```
Comment (enisn) [inline src/CmsKitDemo/CmsKitDemoModule.cs:334]: I think something like `image-gallery/{id}` or `image-gallery/{id}/detail` is much better convention for understanding

File: src/CmsKitDemo/Services/ImageGalleryAppService.cs [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+                Description = x.Description,
+                CoverImageMediaId = x.CoverImageMediaId,
+
+                CommentCount = (from comment in dbContext.Set<Comment>()
+                                where comment.EntityType == CmsKitDemoConsts.ImageGalleryEntityType && comment.EntityId == x.Id.ToString()
+                                select comment).Count(),
+
+                LikeCount = (from reaction in dbContext.Set<UserReaction>()
+                             where reaction.EntityType == CmsKitDemoConsts.ImageGalleryEntityType && reaction.EntityId == x.Id.ToString()
+                             select reaction).Count()
```
Comment (enisn) [inline src/CmsKitDemo/Services/ImageGalleryAppService.cs:44]: Maybe we can add these methods to CmsKit repositories in the next versions, 

A method exists for reactions but nothing for comment
https://github.com/abpframework/abp/blob/dev/modules/cms-kit/src/Volo.CmsKit.EntityFrameworkCore/Volo/CmsKit/Reactions/EfCoreUserReactionRepository.cs#L59

It's ok for this project ✅

File: src/CmsKitDemo/Services/ImageGalleryAppService.cs [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        public ImageGalleryAppService(IRepository<GalleryImage, Guid> repository) : base(repository)
+        {
+            CreatePolicyName = CmsKitDemoPermissions.GalleryImage.Create;
+            UpdatePolicyName = CmsKitDemoPermissions.GalleryImage.Update;
+            DeletePolicyName = CmsKitDemoPermissions.GalleryImage.Delete;
+        }
+
+        public async Task<List<GalleryImageWithDetailsDto>> GetDetailedListAsync()
+        {
+            var dbContext = await Repository.GetDbContextAsync();
```
Comment (enisn) [inline src/CmsKitDemo/Services/ImageGalleryAppService.cs:27]: I'm not sure using DbContext in AppService

Comment (enisn) [reply src/CmsKitDemo/Services/ImageGalleryAppService.cs:27]: Since it's no-layer app, it's ok ✅

Comment (EngincanV) [reply src/CmsKitDemo/Services/ImageGalleryAppService.cs:27]: I wanted to keep it simple and therefore didn't create a custom repository. 👍

File: src/CmsKitDemo/CmsKitDemoModule.cs (hunk 1 above)
Comment (EngincanV) [reply src/CmsKitDemo/CmsKitDemoModule.cs:334]: Agree, I'll update the page route ✅

**Review by enisn — APPROVED** [review]

**Pull request merged**